File: js/task2/S2_FOR_03.js
// S2_FOR_03 — Sum until threshold

function sumUntil(nums, threshold) {
  let sum = 0;

  for (const n of nums) {
    if (sum + n > threshold) {
      break;
    }
    sum += n;
  }

  return sum;
}

// Tests
console.log(sumUntil([1, 2, 3, 4], 5));   // 3 (1+2, stop before 6)
console.log(sumUntil([2, 2, 2], 10));     // 6
console.log(sumUntil([5, 5, 5], 7));      // 5
console.log(sumUntil([], 10));            // 0

File: python/task2b/S2_FOR_03.py
# S2_FOR_03 — Sum until threshold

def sum_until(nums, threshold):
    total = 0
    for n in nums:
        if total + n > threshold:
            break
        total += n
    return total


def main():
    tests = [
        ([1, 2, 3, 4], 5),     # 1+2=3, next 3 would exceed -> 3
        ([5, 5, 5], 10),       # 5+5=10
        ([1, 2, 3], 100),      # sum all
        ([10, 1], 5),          # first already exceeds -> 0
        ([], 10),
    ]

    for nums, t in tests:
        print(f"nums={nums}, threshold={t} -> sum={sum_until(nums, t)}")


if __name__ == "__main__":
    main()
